Carla: update to 2.3.0

jkoderu-git opened the pull request:
@nutcase84

jkoderu-git commented:
carla-control command does not work:
```
Traceback (most recent call last):
File "/usr/share/carla/carla-control", line 22, in <module>
from carla_control import *
File "/usr/share/carla/carla_control.py", line 34, in <module>
from liblo import (
ModuleNotFoundError: No module named 'liblo'
```
liblo-devel is inserted in the template. What is the problem?

fosslinux commented:
Package this: https://github.com/dsacre/pyliblo and make it a dependency.

fosslinux commented:
The template looks mostly ok for pyliblo, but you should put that commit in this PR. Everything that is needed for one package to be built should all be in the same PR.

jkoderu-git commented:
Many thanks for your patience @fosslinux

jkoderu-git commented:
I updated Carla to 2.2.0, which has the qt5 patch included, and fixed xlint for pyliblo. I have 2 questions:
- local xlint does not complain of deprecated `noarch` and Travis does
- the new patch for libmagic fails to apply

ericonr commented:
On Travis we run the git version of xlint.

jkoderu-git commented:
I cleaned the commits and removed JUCE used for VST3 plugins to support `x86_64-musl`.

`aarch64` fails with:
```bash
make[1]: Entering directory '/builddir/Carla-2.2.0/source/frontend'
/usr/bin/pyrcc5 ../../resources/resources.qrc -o resources_rc.py
/usr/bin/pyrcc5: line 2: /usr/bin/python3: No such file or directory
make[1]: *** [Makefile:105: resources_rc.py] Error 127
make[1]: Leaving directory '/builddir/Carla-2.2.0/source/frontend'
make: *** [Makefile:175: frontend] Error 2
```

`armv7l` has a dependency problem:
```bash
=> libedit-20191231.3.1_1: building [gnu-configure] (dependency of pcre) for x86_64...
   [target] ncurses-devel-6.2_3: found (/host/binpkgs)
   [runtime] ncurses-devel-6.2_3: found (/host/binpkgs)
   [runtime] libedit-20191231.3.1_1: not found (self, ignored)
=> libedit-20191231.3.1_1: installing target dependencies: ncurses-devel-6.2_3 ...
=> libedit-20191231.3.1_1: removing autodeps, please wait...
=> ERROR: libedit-20191231.3.1_1: failed to install 'ncurses-devel-6.2_3' dependency! (error 19)
ncurses-libs-6.2_3 in transaction breaks installed pkg `ncurses-6.2_2'
```

I do not know how to fix it, I leave it to someone interested.

fosslinux commented:
The arm problem would have been a transient issue The aarch64 issue can be resolved by putting python3 in hostmakdependes

ericonr commented:
@jkoderu-git ping?

ericonr commented on srcpkgs/Carla/template:
```diff
@@ -1,7 +1,7 @@
 pkgname=Carla
-version=2.1
-revision=2
+version=2.2.0
```
2.3.0 is out

ericonr commented on srcpkgs/Carla/template:
```diff
@@ -10,17 +10,18 @@ makedepends="python3-PyQt5 libmagic file-devel libsndfile-devel
 
 case $XBPS_TARGET_MACHINE in
-	x86_64* | i686*);;
+	x86_64 | i686);;
+	x86_64-musl) make_build_args+=" USING_JUCE=false";;
```
Please add a comment explaining why this is necessary.

jkoderu-git commented:
@fosslinux Thank you, I added it now.

@ericonr I resolved the requested changes. Github CI works but I can't test locally because of `=> ERROR: Build-time cyclic dependency perl(host), perl-AnyEvent(host), perl(host) detected.`

ericonr commented:
Hi, I think you closed this accidentally?

jkoderu-git commented:
> Hi, I think you closed this accidentally?

Hi, yes. I had problems with my local git repository.

ericonr commented on srcpkgs/pyliblo/template:
```diff
@@ -0,0 +1,17 @@
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-Cython"
+makedepends="python3-devel liblo-devel"
+depends="liblo"
```
SONAME: liblo.so.7 <-> liblo>=0.26_1

This was detected automatically, so the `depends` entry isn't necessary.

ericonr reviewed (commented):
Please squash the Carla commits into a single one.

ericonr commented on srcpkgs/Carla/template:
```diff
@@ -10,19 +10,18 @@ makedepends="python3-PyQt5 libmagic file-devel libsndfile-devel
-	x86_64* | i686*);;
-	*) make_build_args+=" NOOPT=true"
-	   make_install_args+=" NOOPT=true";;
+	# No musl support for JUCE
+	*-musl) make_build_args+=" USING_JUCE=false";;
```
What was the error?

jkoderu-git replied:
https://github.com/jkoderu-git/void-packages/runs/2478321591?check_suite_focus=true

ericonr replied:
Interesting, it did work for `x86_64-musl`... So there's something wrong in the cross build for musl.

jkoderu-git replied:
I think I used another case when this PR was open first:
```
        x86_64 | i686);;
	x86_64-musl) make_build_args+=" USING_JUCE=false";;
```

I tested again without and it does not work for `x86_64-musl` either.
https://github.com/jkoderu-git/void-packages/pull/2

ericonr replied:
Ok, errors happen here:

```
Compiling juce_audio_processors.cpp
In file included from ../../includes/vst3sdk/pluginterfaces/base/ftypes.h:19,
                 from ../../includes/vst3sdk/pluginterfaces/base/fstrdefs.h:19,
                 from ../../includes/vst3sdk/pluginterfaces/vst/vsttypes.h:19,
                 from format_types/juce_VST3Headers.h:102,
                 from format_types/juce_VST3PluginFormat.cpp:28,
                 from juce_audio_processors.cpp:152:
../../includes/vst3sdk/pluginterfaces/base/fplatform.h:188: warning: ignoring '#pragma error unknown' [-Wunknown-pragmas]
  188 |  #pragma error unknown platform
      | 
In file included from format_types/juce_VST3Headers.h:109,
                 from format_types/juce_VST3PluginFormat.cpp:28,
                 from juce_audio_processors.cpp:152:
../../includes/vst3sdk/base/source/fstring.cpp:1998:2: warning: #warning DEPRECATED No Linux implementation [-Wcpp]
 1998 | #warning DEPRECATED No Linux implementation
      |  ^~~~~~~
In file included from format_types/juce_VST3Headers.h:112,
                 from format_types/juce_VST3PluginFormat.cpp:28,
                 from juce_audio_processors.cpp:152:
../../includes/vst3sdk/base/thread/source/flock.cpp:81:2: warning: #warning implement FLock! [-Wcpp]
   81 | #warning implement FLock!
      |  ^~~~~~~
In file included from format_types/juce_VST3Headers.h:120,
                 from format_types/juce_VST3PluginFormat.cpp:28,
                 from juce_audio_processors.cpp:152:
../../includes/vst3sdk/pluginterfaces/base/funknown.cpp:78:2: warning: #warning implement me! [-Wcpp]
   78 | #warning implement me!
      |  ^~~~~~~
In file included from format_types/juce_VST3Headers.h:122,
                 from format_types/juce_VST3PluginFormat.cpp:28,
                 from juce_audio_processors.cpp:152:
../../includes/vst3sdk/pluginterfaces/base/ustring.cpp:163:2: warning: #warning Implement me [-Wcpp]
  163 | #warning Implement me
      |  ^~~~~~~
../../includes/vst3sdk/pluginterfaces/base/ustring.cpp:201:2: warning: #warning Implement me [-Wcpp]
  201 | #warning Implement me
      |  ^~~~~~~
../../includes/vst3sdk/pluginterfaces/base/ustring.cpp:228:2: warning: #warning Implement me [-Wcpp]
  228 | #warning Implement me
      |  ^~~~~~~
../../includes/vst3sdk/pluginterfaces/base/ustring.cpp:267:2: warning: #warning Implement me [-Wcpp]
  267 | #warning Implement me
      |  ^~~~~~~
Creating juce_audio_basics.a
make[1]: Leaving directory '/builddir/Carla-2.3.0/source/modules/juce_audio_basics'
make[1]: Entering directory '/builddir/Carla-2.3.0/source/modules/juce_core'
Compiling juce_core.cpp
In file included from juce_core.cpp:69:
/usr/aarch64-linux-musl/usr/include/sys/errno.h:1:2: warning: #warning redirecting incorrect #include <sys/errno.h> to <errno.h> [-Wcpp]
    1 | #warning redirecting incorrect #include <sys/errno.h> to <errno.h>
      |  ^~~~~~~
juce_core.cpp:95:12: fatal error: execinfo.h: No such file or directory
   95 |   #include <execinfo.h>
      |            ^~~~~~~~~~~~
compilation terminated.
```

some of the macros weren't enough.

ericonr replied:
I think we can consider merging as is if you add a big `FIXME` comment here about `juce` on musl, but we might be able to fix...

ericonr replied:
Maybe it's due to being run in your fork? Feel free to push it here for CI to run locally :)

jkoderu-git replied:
> I changed the case in my repo to try a fix for musl
> 
> ```
> if [ "$XBPS_TARGET_LIBC" = musl ]; then
> 	makedepends+=" libexecinfo-devel"
> fi
> ```
> 
> but I can't test because of a CI error.
> 
> ```
> Run common/travis/changed_templates.sh
> Changed packages:
> fatal: bad revision ''
> ```

It does not work. I add the `FIXME` then.

paper42 commented on srcpkgs/Carla/template:
```diff
@@ -10,20 +10,19 @@ makedepends="python3-PyQt5 libmagic file-devel libsndfile-devel
-	*) make_build_args+=" NOOPT=true"
-	   make_install_args+=" NOOPT=true";;
-esac
+/* FIXME: JUCE does not compile on musl */
```
`/* */` is not a comment

jkoderu-git replied:
Fixed, thanks!

paper42 commented:
Why are you skipping CI?

jkoderu-git commented:
It already compiled to all machines. I skipped it because I only changed the comment you said to save CI time.

sgn commented:
Hey, please rebase your branch on top of master, as of it's now, your PR is lagging behind too much, the lint steps has problems if it's lagged behind more than 200 changes

jkoderu-git commented:
> Hey, please rebase your branch on top of master, as of it's now, your PR is lagging behind too much, the lint steps has problems if it's lagged behind more than 200 changes

Thank you!

paper42 commented:
2.3.1 is out, could you rebase again on top of master?

paper42 commented on srcpkgs/pyliblo/template:
```diff
@@ -0,0 +1,16 @@
+pkgname=pyliblo
+version=0.10.0
+revision=1
+wrksrc="pyliblo-${version}"
```
this is the default value of $wrksrc, so it can be dropped

ericonr commented:
Supersed by #33809, thanks!
